crates/ty_python_semantic/src/types.rs

```diff
@@ -9389,7 +9389,17 @@ impl<'db> BoundSuperType<'db> {
             ));
         }
 
-        let pivot_class = ClassBase::try_from_type(db, pivot_class_type).ok_or({
+        // TODO: having to get a class-literal just to pass it in here is silly.
+        // `BoundSuperType` should use a different enum rather than reusing `ClassBase`.
```

Comment:
I'm not convinced of this comment; I think the "pivot class" of a bound `super` object really is a "base class" (that is, it is anything that can exist in an MRO), and it's good to represent it with `ClassBase`.

I would frame the problem here slightly differently: `ClassBase` enum should represent "a thing that can be in an MRO", but the constructor `ClassBase::try_from_type` is also emulating the behavior of `__mro_entries__`, and there are cases for wanting to construct a "thing that can be in an MRO" from a type without emulating `__mro_entries__`. So perhaps there should be a way to construct a `ClassBase` that doesn't model `__mro_entries__` (and would thus return `None` if you tried to use `NamedTuple` as a base type.)